rotors_gazebo_plugins/src/gazebo_wind_plugin.cpp

```diff
@@ -208,6 +331,109 @@ void GazeboWindPlugin::CreatePubsAndSubs() {
                                            true);
 }
 
+void GazeboWindPlugin::ReadCustomWindField(std::string& custom_wind_field_path) {
+  std::ifstream fin;
+  fin.open(custom_wind_field_path);
+  if (fin.is_open()) {
+    std::string data_name;
+    float data;
+    // Read the line with the variable name.
+    while (fin >> data_name) {
+      // Save data on following line into the correct variable.
```

Comment:
Can you put this in a parsing function and use `switch` instead of `if`, `else if`.